app/models/concerns/has_file.rb

```diff
@@ -11,13 +11,29 @@ def file_mandatory?
     true
   end
 
+  def image_path(version = nil)
+    self.file_url(version).to_s.gsub(/^#{Regexp.escape(CONFIG['storage']['endpoint'])}/, CONFIG['storage']['public_endpoint'])
+  end
+
+  def file_path
+    self.image_path
+  end
+
+  def embed_path
+    self.image_path('embed')
+  end
+
   module ClassMethods
     def max_size
```

Comment:
This method is confusing... can we implement one in `UploadedImage` and another one in `UploadedVideo`?

Reply:
Done